Page object branch

NataSelenium opened the pull request:
Hi, please review my files for task 60

Commit:
Adde new file

Commit:
I deleted file

Commit:
Added new files

Commit:
Modified Data Driven Test files

Commit:
Modified DDT files

Commit:
Modified files 6th Apr

Commit:
Added tests for task 60

Commit:
Remove files

Commit:
Add Page Factory tests

Commit:
Add screenshot test

EugenBorisik commented on PageObjectTest/PageObjects/InboxPage.cs:
```diff
@@ -0,0 +1,24 @@
+{
+    class InboxPage
+    {
+        public string exitCssSelector = "a[title='Sign out']";
```
Make them constants

EugenBorisik replied:
All fields should be private

EugenBorisik commented on PageObjectTest/PageObjects/InboxPage.cs:
```diff
@@ -0,0 +1,24 @@
+
+namespace SeleniumTask1.PageObjectTest.PageObjects
+{
+    class InboxPage
```
Create ctor, which gets IWebDriver

EugenBorisik commented on PageObjectTest/PageObjects/InboxPage.cs:
```diff
@@ -0,0 +1,24 @@
+        public string exitCssSelector = "a[title='Sign out']";
+        public string expectedTitle = "RMSys - Sign In";
+
+        public void FindWebElementAndClick(IWebDriver driver, string exit)
```
With ctor you will not need to pass IWebDriver everytime

EugenBorisik replied:
Explain, why do you need such strange method?

EugenBorisik commented on PageObjectTest/PageObjects/InboxPage.cs:
```diff
@@ -0,0 +1,24 @@
+            driver.FindElement(By.CssSelector(exit)).Click();
+        }
+
+        public void WaitExpectedConditionWebElement(IWebDriver driver, string exit)
```
Explain, why do you need such strange method?

EugenBorisik commented on PageObjectTest/PageObjects/LoginPage.cs:
```diff
@@ -0,0 +1,53 @@
+    class LoginPage
+    {
+        public IWebDriver driver;
+        public string loginId = "Username";
```
Make them constants

EugenBorisik commented on PageObjectTest/PageObjects/LoginPage.cs:
```diff
@@ -0,0 +1,53 @@
+
+        public void GetPage(string url)
+        {
+            //Go to the URL
```
Remove comments

EugenBorisik commented on PageObjectTest/PageObjects/LoginPage.cs:
```diff
@@ -0,0 +1,53 @@
+    {
+        public IWebDriver driver;
+        public string loginId = "Username";
+        public string loginValue = "nataliadamorad";
```
All fields should be private

EugenBorisik commented on PageObjectTest/PageObjects/LoginPage.cs:
```diff
@@ -0,0 +1,53 @@
+            driver.Manage().Window.Maximize();
+        }
+
+        public void FindWebElementAndSendKeys(string id, string text)
```
Rename all these methods after making fields private

EugenBorisik commented on PageObjectTest/TestCases/LoginTest.cs:
```diff
@@ -0,0 +1,39 @@
+            //Find the 'password' input and type the password
+            loginPage.FindWebElementAndSendKeys(loginPage.passwordId, loginPage.passwordValue);
+            //Find the 'Submit' button and click
+            loginPage.FindWebElementAndClick(loginPage.submitButtonId);
```
Collect these 3 methods into one named "login(string username, string password)"

EugenBorisik commented on PageObjectTest/TestCases/LogoutTest.cs:
```diff
@@ -0,0 +1,48 @@
+            //Find the 'Submit' button and click
+            loginPage.FindWebElementAndClick(loginPage.submitButtonId);
+            //Wait when title becomes 'RMSys - Home' - about 1 min with my test
+            loginPage.WaitExpectedConditionTitle(seleniumDriver, loginPage.expectedTitle);
```
All waiters should be on page object layer

EugenBorisik reviewed (changes requested):
@NataSelenium please update code according to comments. Update branch with page factory accordingly

Commit:
Update Page Object Tests

NataSelenium commented:
Hi😊
I updated code. I added those strange methods, because I believe ‘the larger the number, the better the result’ 😐
Now I understand my decision complicated code ☹
Thanks

EugenBorisik commented on PageObjectTest/PageObjects/InboxPage.cs:
```diff
@@ -0,0 +1,20 @@
+{
+    class InboxPage
+    {
+        IWebDriver driver;
```
Add access modifier

EugenBorisik commented on PageObjectTest/PageObjects/LoginPage.cs:
```diff
@@ -0,0 +1,36 @@
+{
+    public class LoginPage
+    {
+        IWebDriver driver;
```
Add access modifiers

EugenBorisik commented on PageObjectTest/TestCases/LoginTest.cs:
```diff
@@ -0,0 +1,39 @@
+    [TestClass]
+    public class LoginClassicTest
+    {
+        IWebDriver seleniumDriver;
```
Add access modifiers

EugenBorisik commented on PageObjectTest/TestCases/LoginTest.cs:
```diff
@@ -0,0 +1,39 @@
+            LoginPage loginPage = new LoginPage(seleniumDriver);
+            loginPage.GetPage();
+            loginPage.Login(loginValue, passwordValue);
+            var wait = new WebDriverWait(seleniumDriver, TimeSpan.FromSeconds(2));
```
Waiters should be in page objects, move this waiter into Login method

EugenBorisik commented on PageObjectTest/TestCases/LoginTest.cs:
```diff
@@ -0,0 +1,39 @@
+        [TestMethod, TestCategory("Page Object Classic")]
+        public void LoginClassicTestMethod()
+        {
+            seleniumDriver = new ChromeDriver();
```
It makes sense to initialize WebDriver instance in before hook

EugenBorisik commented on PageObjectTest/TestCases/LogoutTest.cs:
```diff
@@ -0,0 +1,42 @@
+            loginPage.GetPage();
+            loginPage.Login(loginValue,passwordValue);
+            InboxPage inboxPage = new InboxPage(seleniumDriver);
+            var wait = new WebDriverWait(seleniumDriver, TimeSpan.FromSeconds(2));
```
Move waiter into page object

EugenBorisik commented on PageObjectTest/TestCases/LogoutTest.cs:
```diff
@@ -0,0 +1,42 @@
+    [TestClass]
+    public class LogoutClassicTest
+    {
+        IWebDriver seleniumDriver;
```
Add access modifiers

EugenBorisik commented on PageObjectTest/TestCases/LogoutTest.cs:
```diff
@@ -0,0 +1,42 @@
+        [TestMethod, TestCategory("Page Object Classic")]
+        public void LogoutClassicTestMethod()
+        {
+            seleniumDriver = new ChromeDriver();
```
Use before hook

EugenBorisik reviewed (changes requested):
@NataSelenium please update code

Commit:
Update Page Object Tests
